client.go

```diff
@@ -208,11 +209,12 @@ func (c *client) unassignRoles(ctx context.Context, roleIDs []string) error {
 
 	for _, id := range roleIDs {
 		if _, err := c.provider.DeleteRoleAssignmentByID(ctx, id); err != nil {
-			detailedErr := new(autorest.DetailedError)
 			// If a role was deleted manually then Azure returns a error and status 204
-			if errors.As(err, detailedErr) && (detailedErr.StatusCode == http.StatusNoContent || detailedErr.StatusCode == http.StatusNotFound) {
-				continue
-			}
+			// @TODO figure out how to get the status code from MSGraph SDK errors; removing autorest for now
```

Comment:
Opted to remove autorest for the time being, but not sure how to best replicate this with the tools at hand. I think the error returned from MSGraph should contain a status code to perform the logic below, but need to double check this

Reply:
I think you can do something similar to what we did in TFVP: https://github.com/hashicorp/terraform-provider-vault/pull/2050/files#diff-de5e35945f06f622d0b58ae2f696ead44ed39d0acb73fc826309a03ce4e00b83R252-R262

Which used https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/MIGRATION_GUIDE.md#error-handling as the guide

Reply:
Not sure if that will work with MSGraph?

Reply:
Nice, it looks to be in the `azcore` library, so I think it should work (🤞🏼) Thanks for the reference 🙏🏼